relayer/relays/parachain/beefy-listener.go

```diff
@@ -602,18 +617,53 @@ func (li *BeefyListener) scanForCommitments(
 					return nil, fmt.Errorf("basic channel commitment hash in digest item does not match the one in the Committed event")
 				}
 
-				bundle := events.Basic.Bundle
+				// Only consider message bundles for the account we're interested in
+				bundles := events.Basic.Bundles
+				var bundle BasicOutboundChannelMessageBundle
+				var bundleFound bool
+				var bundleIndex int
+				for i, b := range bundles {
+					if b.Account == accountAsArray {
+						bundleFound = true
+						bundleIndex = i
+						break
+					}
+				}
+				if bundleFound {
+					bundle = bundles[bundleIndex]
+				} else {
+					continue
+				}
+
+				// Fetch Merkle proof for this bundle
+				api, err := gsrpc.NewSubstrateAPI(li.config.Parachain.Endpoint)
+				if err != nil {
+					return nil, fmt.Errorf("create client: %w", err)
+				}
+
+				var proofHex string
+				err = api.Client.Call(&proofHex, "basicOutboundChannel_getMerkleProof", digestItem.AsCommitment.Hash.Hex(), bundleIndex)
```

Comment:
An initialized api already exists on the listener object:
```suggestion
				err = li.parachainConnection.API().Client.Call(&proofHex, "basicOutboundChannel_getMerkleProof", digestItem.AsCommitment.Hash.Hex(), bundleIndex)
```